zfsbootmenu/lib/zfsbootmenu-core.sh

```diff
@@ -1673,6 +1673,19 @@ load_key() {
     return 0
   fi
 
+  # Run early load_key hooks, if they exist
+  if [ -d /libexec/load_key.d ]; then
+    for _hook in /libexec/load_key.d/*; do
+      zinfo "Processing hook: ${_hook}"
+      [ -x "${_hook}" ] || continue
+      if ! "${_hook}" "${encroot}"; then
+        zwarn "load-key hook failed for '${encroot}', aborting load attempt"
```

Comment:
This warning should indicate the name of the hook that failed, otherwise the user is left guessing if multiple `load_key.d` hooks exist.

Reply:
Agree. Put "${_hook}" into zwarn

Reply:
This change has put a double-quoted argument inside another double-quoted argument, breaking it. 

Are any of these changes being tested prior to committing them? I'd like to see this PR merged, but the current process you're using is unacceptable. Look through all of the open conversations/review notes, fix everything addressed in them, and squash everything down to a single commit. Only when that's all done and you're confident that everything has been fully address, push to the PR so that there's a single commit ready for final review.

Reply:
You right. It was my mistake with double-quoted argument and I fixed that. I've just tested the final version. It works. I can delete this pr and open a new one with all the commits squashed  to one. Is that what you have in mind?

Reply:
No, don't close a PR and open the same thing. Use `git rebase` or other git history rewriting tricks to collapse all of these edits and force-push back to this branch.

Reply:
Sorry I'm not an experienced user of a github. This is my first pr. From my point of view all the issues and notes you addressed are implemented and there are no open discussions other then the one with the releng/rst2help.sh doc generator which is going to produce a bunch of a new files. As I've read in some articles, there is no way to rewrite the history in github. There is a possibility to create a new commits that will restore the previous state of the data by changing the current data to the old one. If this is true, then I have no idea how to collapse the topic, like this one? What should be the result of such a collapse? In my judgement, as far as 90% of these topics relates to the hook script that finally was excluded from the pr completely, it is not a bad idea to make a new pr with only few files in it and with very little changes in them. The hook itself is quite easy, isn't it? It would be easy to review these files and accept the pr.

Reply:
you probably shouldn't be using github's web editor to make these changes, but instead clone your fork, commit to the branch locally, and push. `git rebase` is a tool within git that allows you to rewrite the history. After rebasing, you can to `git push --force` to update the PR. There are a number of good tutorials on git concepts like rebasing online.

this is not a "github" thing but a git thing. github is just a place to collaborate using git.

Reply:
Ok, finally I've managed to make a `git rebase` and  then`git push --force` Hope, the result is what you had in mind